src/Http/Http/src/WriteOnlyPipeStream.cs

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO.Pipelines
+{
+    /// <summary>
+    /// Represents a WriteOnlyStream backed by a PipeWriter
+    /// </summary>
+    public class WriteOnlyPipeStream : Stream
```

Comment:
Most of this is copied from: https://github.com/aspnet/KestrelHttpServer/blob/master/src/Kestrel.Core/Internal/Http/HttpResponseStream.cs